### ISSUE TITLE
`inline_trait_bounds` should not be triggered on `#[automatically_derived]` code

### ISSUE THREAD

**Issue by nazar-pc** [rust-lang/rust-clippy#17110]
### Summary

A lot of derive macros produce code that is outside of developer's control.

While `clippy::inline_trait_bounds` is useful, it should not be triggered for generated code since it may be impossible to convince authors of derive macros to change it.

### Lint Name

inline_trait_bounds

### Version

```text
rustc 1.98.0-nightly (cced03bfd 2026-05-28)
binary: rustc
commit-hash: cced03bfd61a304243a34504618ecec86c17063f
commit-date: 2026-05-28
host: x86_64-unknown-linux-gnu
release: 1.98.0-nightly
LLVM version: 22.1.6
```

**Comment by samueltardieu** [comment]
Note: the fix you had to do in `parity-scale-codec` is probably because an improper usage of `quote_spanned!`, which prevents the compiler and Clippy from seeing the expanded code as being from an external macro.

**Comment by nazar-pc** [comment]
Hm, it did produce trait implementations with `#[automatically_derived]`, so I'm not sure what else should be done there.

**Comment by samueltardieu** [comment]
See an explanation of the problems with `quote_spanned!` here: https://rust-lang.zulipchat.com/#narrow/channel/131828-t-compiler/topic/Incorrect.20compiler.20fixes.20in.20presence.20of.20bad.20proc.20macros

**Comment by samueltardieu** [comment]
> Hm, it did produce trait implementations with `#[automatically_derived]`, so I'm not sure what else should be done there.

I've added code to check for this in the linked PR.
